test/performance/performance_test.go

```diff
@@ -0,0 +1,325 @@
+//go:build performance
+// +build performance
+
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package performance
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/serving/pkg/apis/serving"
+
+	"golang.org/x/sync/errgroup"
+	kperf "knative.dev/kperf/pkg"
+	"knative.dev/kperf/pkg/command/service"
+	"knative.dev/pkg/signals"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+	"knative.dev/serving/test"
+	v1test "knative.dev/serving/test/v1"
+)
+
+const (
+	ServiceNamePrefix    = "ksvc"
+	ServicesCountEnvVar  = "KPERF_SERVICES_COUNT"
+	DefaultServicesCount = 10
+)
+
+func getServicesCount() (int, error) {
+	servicesCount := os.Getenv(ServicesCountEnvVar)
+	if servicesCount == "" {
+		return 10, nil
+	}
+	intVar, err := strconv.Atoi(servicesCount)
+	if err != nil {
+		return 0, err
+	}
+	return intVar, nil
+}
+
+func TestPerformanceServiceCreation(t *testing.T) {
+	clients := test.Setup(t)
+
+	servicesCount, err := getServicesCount()
+	if err != nil {
+		t.Fatalf("Failed to get services count %#v", err)
+	}
+	servicesRange := fmt.Sprintf("1,%d", servicesCount-1)
+
+	namespaces := []string{}
+	nsNamePrefix := test.AppendRandomString("kperf")
+	t.Log("Creating namespaces to be used by kperf.")
+	for i := 1; i < servicesCount; i++ {
```

Comment:
Does this actually create `serviceCount - 1` services (starting from 1) ?

Reply:
fixed